k8s/prometheus-operator/values.yaml

```diff
@@ -122,31 +113,38 @@ kubeEtcd:
     insecureSkipVerify: true
     https: false
 kubeStateMetrics:
-  enabled: false
+  enabled: true
+kube-state-metrics:
+
 prometheus:
   prometheusSpec:
     serviceMonitorSelectorNilUsesHelmValues: false
     affinity:
       podAntiAffinity:
         requiredDuringSchedulingIgnoredDuringExecution:
-        - labelSelector:
-            matchExpressions:
-            - key: app
-              operator: In
-              values:
-              - redis
-          topologyKey: "kubernetes.io/hostname"
+          - topologyKey: "kubernetes.io/hostname"
+            labelSelector:
+              matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                    - redis
       podAffinity:
         requiredDuringSchedulingIgnoredDuringExecution:
-        - labelSelector:
-            matchExpressions:
-            - key: app
-              operator: In
-              values:
-              - prometheus-operator-operator
-          topologyKey: "kubernetes.io/hostname"
+          - topologyKey: "kubernetes.io/hostname"
+            labelSelector:
+              matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                    - prometheus-operator-operator
     nodeSelector:
       testground.nodetype: infra
+    disableCompaction: true
+    query:
+      maxSamples: 187500000
```

Comment:
Each sample is 16 bytes, so this is 6000mi / 16 / 2

Reply:
I imagine we might hit OOM issues because of queries in the future, but this is not the case right now as far as I can tell.